server/src/api/observation/observation.controller.ts

```diff
@@ -1,30 +1,65 @@
 import { Request, Response } from 'express';
-import { getManager } from 'typeorm';
 import { ObservationApiModel } from './observation.types';
 import { Observation } from './observation.model';
 import { ApiRequest, EdipiParam } from '../api.router';
+import { Log } from '../../util/log';
+import { ReportSchema } from '../report-schema/report-schema.model';
+import { RosterInfo, getRostersForIndividual } from '../roster/roster.controller';
 import { timestampColumnTransformer } from '../../util/util';
 
+
 class ObservationController {
   async getAllObservations(req: Request, res: Response) {
     return res.json(await Observation.find());
   }
 
   async createObservation(req: ApiRequest<EdipiParam, ObservationApiModel>, res: Response) {
-    return res.json(await createObservation(req));
+
+    Log.info('Creating observation', req.body);
+
+    const reportingGroup = req.body.reportingGroup;
+    const typeId = req.body.typeId;
+
+    const rosters: RosterInfo[] = await getRostersForIndividual(req.body.edipi, req.body.timestamp.toString(10));
+
+    for (const roster of rosters) {
+      if (hasReportingGroup(roster, reportingGroup)) {
+        const reportSchema = await findReportSchema(typeId, roster.unit.org?.id);
+        if (reportSchema) {
+          return res.json(saveObservationWithReportSchema(req, reportSchema));
+        }
+      }
+    }
+    Log.error(`Unable to save observation from: ${req.body}`);
+    return res.json(new Observation());
+
   }
 }
 
-async function createObservation(req: ApiRequest<EdipiParam, ObservationApiModel>) {
-  // getManager().query() does not return the inserted row
-  console.log(`Creating observation ${JSON.stringify(req.body)}`);
-  await getManager()
-    .query(
-      'INSERT INTO public.observation(document_id, edipi, timestamp, unit_id, unit, type_id, type_org)'
-      + 'VALUES ($1, $2, $3, $4, $5, $6, $7)',
-      [req.body.documentId, req.body.edipi, timestampColumnTransformer.to(req.body.timestamp),
-        req.body.unitId, req.body.unit, req.body.typeId, req.body.orgId],
-    );
+function hasReportingGroup(rosterInfo: RosterInfo, reportingGroup: string | undefined) {
+  return reportingGroup && rosterInfo.unit.org?.reportingGroup === reportingGroup;
+}
+
+async function findReportSchema(reportSchemaId: string, orgId: number | undefined) {
+  return ReportSchema.findOne({
+    relations: ['org'],
```

Comment:
I don't think you need this relation here. It doesn't appear to be used anywhere. I don't think it's necessary to query on the `org` property, but I could be wrong.

Reply:
I took this from an existing example. Each ORM is different. ORM knowledge is not directly transferable from one ORM to other given all the differences. ORM is convenient but makes us to guess the code, it is a good PR discussion started, it makes us dumber. :-)

Reply:
Removal of `relations: ['org'],` did not work.

Reply:
That's fine we can leave it.

Just out of curiosity, what happened when you removed it that stopped things from working?